server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/test/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutesTest.java

```diff
@@ -261,6 +270,62 @@ void userVaultAPIShouldReturnInvalidWhenPassingCaseInsensitiveAction() {
                 .body("details", is(notNullValue()));
         }
 
+        @Test
+        void purgeAPIShouldReturnInvalidWhenActionIsMissing() {
+            when()
+                .post()
+            .then()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .body("statusCode", is(400))
+                .body("type", is(ErrorResponder.ErrorType.INVALID_ARGUMENT.getType()))
+                .body("message", is("Invalid arguments supplied in the user request"))
+                .body("details", is("action parameter is missing"));
+        }
+
+        @Test
+        void purgeAPIShouldReturnInvalidWhenPassingEmptyAction() {
+            given()
+                .queryParam("action", "")
+            .when()
+                .post()
+            .then()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .body("statusCode", is(400))
+                .body("type", is(ErrorResponder.ErrorType.INVALID_ARGUMENT.getType()))
+                .body("message", is("Invalid arguments supplied in the user request"))
+                .body("details", is("action cannot be empty or blank"));
+        }
+
+        @Test
+        void purgeAPIShouldReturnInvalidWhenActionIsInValid() {
+            given()
+                .queryParam("action", "invalid action")
+            .when()
+                .post()
+            .then()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .body("statusCode", is(400))
+                .body("type", is(ErrorResponder.ErrorType.INVALID_ARGUMENT.getType()))
+                .body("message", is("Invalid arguments supplied in the user request"))
+                .body("details", is(String.format("'invalid action' is not a valid action. Supported values are: (%s)",
```

Comment:
please assert only on a substring of the message, not on the list of possible actions, and write a specific test to check the list of possible value.